Hotfix anvilprod: Fix: Access denied to two anvil8 snapshots (#6925)

dsotirho-ucsc opened the pull request:
<!--
This is the PR template for hotfix PRs against `anvilprod`.
-->

Connected issue: #6925

## Notes:

- Requires a deindex of the following snapshots from `anvil8` and `anvil8-it`:
    - `tdr:bigquery:gcp:datarepo-d36f3ea3:ANVIL_CCDG_WashU_CVD_EOCAD_WashU_CAD_GRU_IRB_WGS_20230525_ANV5_202409302305:`
    - `tdr:bigquery:gcp:datarepo-b36de71b:ANVIL_ccdg_broad_ai_ibd_daly_mcgovern_share_wes_20240104_ANV5_202409261554:`

## Checklist


### Author

- [x] Target branch is `anvilprod`
- [x] Name of PR branch matches `hotfixes/<GitHub handle of author>/<issue#>-<slug>-anvilprod`
- [x] On ZenHub, PR is connected to the issue it hotfixes
- [x] PR description links to connected issue
- [x] PR title is `Hotfix anvilprod: ` followed by title of connected issue
- [x] PR title references the connected issue


### Author (hotfixes)

- [x] Added `h` tag to commit title <sub>or this PR does not include a temporary hotfix</sub>
- [x] Added `H` tag to commit title <sub>or this PR does not include a permanent hotfix</sub>
- [x] Added `hotfix` label to PR
- [x] This PR is labeled `partial` <sub>or represents a permanent hotfix</sub>


### Author (before every review)

- [x] Rebased PR branch on `anvilprod`, squashed old fixups
- [x] Ran `make requirements_update` <sub>or this PR does not modify `requirements*.txt`, `common.mk`, `Makefile` and `Dockerfile`</sub>
- [x] Added `R` tag to commit title <sub>or this PR does not modify `requirements*.txt`</sub>
- [x] This PR is labeled `reqs` <sub>or does not modify `requirements*.txt`</sub>


### System administrator (after approval)

- [ ] Actually approved the PR
- [ ] Decided if PR can be labeled `no sandbox`
- [ ] A comment to this PR details the completed security design review
- [ ] PR title is appropriate as title of merge commit
- [ ] Moved connected issue to *Approved* column
- [ ] PR is assigned to only the operator


### Operator (before pushing merge the commit)

- [ ] Squashed PR branch and rebased onto `anvilprod`
- [ ] Sanity-checked history
- [ ] Pushed PR branch to GitHub
- [ ] Added `sandbox` label <sub>or PR is labeled `no sandbox`</sub>
- [ ] Pushed PR branch to GitLab `anvilprod` <sub>or PR is labeled `no sandbox`</sub>
- [ ] Build passes in `hammerbox` deployment <sub>or PR is labeled `no sandbox`</sub>
- [ ] Reviewed build logs for anomalies in `hammerbox` deployment <sub>or PR is labeled `no sandbox`</sub>
- [ ] The title of the merge commit starts with the title of this PR
- [ ] Added PR # reference to merge commit title
- [ ] Collected commit title tags in merge commit title <sub>but excluded any `p` tags</sub>
- [ ] Moved connected issue to *Merged stable* column in ZenHub
- [ ] Pushed merge commit to GitHub

### Operator (after pushing the merge commit)

- [ ] Pushed merge commit to GitLab `anvilprod`
- [ ] Build passes on GitLab `anvilprod`
- [ ] Reviewed build logs for anomalies on GitLab `anvilprod`
- [ ] Deleted PR branch from GitHub
- [ ] Deleted PR branch from GitLab `anvilprod`


### Operator (reindex)

- [ ] Deindexed all unreferenced catalogs in `anvilprod` <sub>or this PR is neither labeled `reindex:partial` nor `reindex:anvilprod`</sub>
- [ ] Deindexed specific sources in `anvilprod` <sub>or this PR is neither labeled `reindex:partial` nor `reindex:anvilprod`</sub>
- [ ] Updated [TDR Datasets and snapshots](https://docs.google.com/spreadsheets/d/1yYiB3TrX0wvLCHvtWY1GTZZcAf5MmMREgn52OKCHsqs/edit?gid=187179274#gid=187179274) spreadsheet
- [ ] Indexed specific sources in `anvilprod` <sub>or this PR is neither labeled `reindex:partial` nor `reindex:anvilprod`</sub>
- [ ] Started reindex in `anvilprod` <sub>or neither this PR nor a failed, prior promotion requires it</sub>
- [ ] Checked for, triaged and possibly requeued messages in both fail queues in `anvilprod` <sub>or neither this PR nor a failed, prior promotion requires it</sub>
- [ ] Emptied fail queues in `anvilprod` <sub>or neither this PR nor a failed, prior promotion requires it</sub>
- [ ] Restarted the Data Browser pipeline for the [ucsc/anvil/anvilprod branch](https://gitlab.explore.anvilproject.org/ucsc/data-browser/-/pipelines/new?ref=ucsc%2Fanvil%2Fanvilprod) on GitLab in `anvilprod` <sub>or neither this PR nor a failed, prior promotion requires it</sub>
- [ ] Restarted `deploy_browser` job in the GitLab pipeline for this PR in `anvilprod` <sub>or neither this PR nor a failed, prior promotion requires it</sub>
- [ ] Created backport PR and linked to it in a comment on this PR


### Operator

- [ ] PR is assigned to no one


## Shorthand for review comments

- `L` line is too long
- `W` line wrapping is wrong
- `Q` bad quotes
- `F` other formatting problem

dsotirho-ucsc committed:
[r H] Fix: Access denied to two anvil8 snapshots (#6925)

dsotirho-ucsc commented:
[6926_IT_2025-02-24.txt](https://github.com/user-attachments/files/18951397/6926_IT_2025-02-24.txt)

hannes-ucsc reviewed (changes requested):
From the attached test logs it appears this was tested with a personal deployment that's based on HCA. If that's indeed the case, it would not represent a sufficiently representative test. How did you ensure that these changes indeed fix or work around  the connected issue?

dsotirho-ucsc committed:
fixup! [r H] Fix: Access denied to two anvil8 snapshots (#6925)

dsotirho-ucsc commented:
This was originally tested only on my personal deployment. PR has now been tested on `hammerbox` and the reindex completed successfully. https://gitlab.explore.anvilproject.org/ucsc/azul/-/pipelines/18243

hannes-ucsc reviewed (changes requested):
I think the hot fix should appear on the promotion branch, and there shouldn't be a separate PR for it.

Also be sure to update the spreadsheet accordingly.

hannes-ucsc commented on deployments/anvilprod/environment.py:
```diff
@@ -895,7 +894,7 @@ def mkdict(previous_catalog: dict[str, str],
     mksrc('bigquery', 'datarepo-f11a2ad0', 'ANVIL_ccdg_broad_ai_ibd_daly_mccauley_wes_20240104_ANV5_202409242021'),
     mksrc('bigquery', 'datarepo-2b74b327', 'ANVIL_ccdg_broad_ai_ibd_daly_mcgovern_niddk_wes_20240104_ANV5_202409242032'), # noqa E501
-    mksrc('bigquery', 'datarepo-b36de71b', 'ANVIL_ccdg_broad_ai_ibd_daly_mcgovern_share_wes_20240104_ANV5_202409261554'), # noqa E501
+    mksrc('bigquery', 'datarepo-b36de71b', 'ANVIL_ccdg_broad_ai_ibd_daly_mcgovern_share_wes_20240104_ANV5_202409261554', pop), # noqa E501
```
I think you should pop the previous snapshot for that dataset. It would be confusing to use a new snapshot for popping an old one.

dsotirho-ucsc commented:
> I think the hot fix should appear on the promotion branch, and there shouldn't be a separate PR for it.
> 
> Also be sure to update the spreadsheet accordingly.

Added hotfix commit f3d3a77 to `anvilprod` promotion  branch (PR #6913). Added notes to PR about which snapshots to deindex and CL item to update spreadsheet.

dsotirho-ucsc commented:
Closing PR in favor of hotfix commit added to `anvilprod` promotion branch.
